pkg/controller/resourcequota/resource_quota_controller.go

```diff
@@ -421,7 +422,9 @@ func (rq *ResourceQuotaController) Sync(discoveryFunc NamespacedResourcesFunc, p
 	// Something has changed, so track the new state and perform a sync.
 	oldResources := make(map[schema.GroupVersionResource]struct{})
 	wait.Until(func() {
-		// Get the current resource list from discovery.
+		// Get the current resource list from discovery.  A failure here will (should?) prevent updates to the sync list.
```

Comment:
Is tolerating partial discovery on start but not on resync what we want? I expected them to match. `ErrGroupDiscoveryFailed` errors expose which group/versions failed, if we want to use that info

Reply:
even without inspecting the details of the error, we could do something like this as a simple start. letting the quota controller continue running without using available resource info from discovery doesn't seem great.

```go
// Get the current resource list from discovery.
newResources, err := GetQuotableResources(discoveryFunc)
if err != nil {
	utilruntime.HandleError(err)

	if discovery.IsGroupDiscoveryFailedError(err) && len(newResources) > 0 {
		// In partial discovery cases, don't remove any existing informers, just add new ones
		for k, v := range oldResources {
			newResources[k] = v
		}
	} else {
		// short circuit in non-discovery error cases or if discovery returned zero resources
		return
	}
}
```